### DISCUSSION TITLE
Can a NEAR node apply a filter to transactions

### DISCUSSION THREAD

**Discussion by wackazong** [near/nearcore#8105] [General]
Is it not possible to configure an indexer node in a way that it only listens to and stores transactions, receipts, and state for specific accounts?

**Comment by frol** [comment]
*Disclaimer: It is better to ask such questions on StackOverflow and tag them with `nearprotocol` tag: https://stackoverflow.com/questions/tagged/nearprotocol*

@wackazong [NEAR Indexer Framework](https://github.com/near/nearcore/tree/master/chain/indexer) entirely relies on the nearcore operation which requires storing all the transactions, receipts, and state of all the accounts at least in 5 recent epochs. Thus, when using the Indexer Framework, there is no way to store less than nearcore needs, yet if you use [NEAR Lake Framework](https://docs.near.org/concepts/advanced/near-lake-framework), you don't need to keep the blockchain data locally as all the blocks were dumped to S3 and you can access arbitrary block at any moment in time.
